lib/seccomp-tools/asm/compiler.rb

```diff
@@ -129,10 +139,17 @@ def compile_cmp(op, val, jt, jf)
       end
 
       def label_offset(label)
+        if label.is_a?(Integer)
+          raise ArgumentError, "Loop detected!" if label == 0
+          raise ArgumentError, "Does not support backward jumping." if label < 0
```

Comment:
Turns out label cannot be negative because it would actually fail with a parse error instead. So this entire block will be removed in my next commit